File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>(); 
        // for(int i = 0; i < nums.length; i++) {
        //     map.put(nums[i], i);  
        // }
        for(int i = 0; i < nums.length; i++) {
            int comp = target - nums[i]; 
            if(map.containsKey(comp) && map.get(comp) != i) {
                return new int[]{i, map.get(comp)}; 
            }
            map.put(nums[i], i);  
        }
        return null;
        
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_nums = {}
        result = []
        
        for i in range(len(nums)):
            dict_nums[nums[i]] = i
        
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in dict_nums and dict_nums[complement] != i:
                result.append(i)
                result.append(dict_nums[complement])
                return result